syntax = "proto3";

package sensor;

service SensorService {
  rpc GenerateSensorData (SensorRequest) returns (SensorResponse) {}
}

message SensorRequest {
  double duration = 1;
  double x0 = 2;
  double y0 = 3;
  int32 complexity = 4;
  repeated double xf = 5;
  repeated double yf = 6;
  string signature = 7;
}

message SensorResponse {
  string sensor_data = 1;
  string error = 2;
  string traceback = 3;
  string controller_dump = 4;
}
